### PR TITLE
Add generic annotator for data layout ops

### PR DESCRIPTION
**Pull request by benkli01**
Data layout ops like unsqueeze are not annotated by the quantizer per default which leads to issues down the line. Therefore we add a generic annotator to explicitly annotate those ops.

### CODE REVIEW

**Comment by digantdesai** [comment]
> Data layout ops like unsqueeze are not annotated by the quantizer per default which leads to issues down the line

Can you elaborate? what issues? Is it just lowering issues?

File: backends/arm/test/quantizer/test_generic_annotater.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+    def forward(self, x):
+        return self.op(x, **self.op_kwargs)
+
+    def example_inputs(self):
+        return self._example_input
+
+
+class TestGenericAnnotator(unittest.TestCase):
+    def check_annotation(self, model):
```
Comment (digantdesai) [inline backends/arm/test/quantizer/test_generic_annotater.py:31]: nice!

**Review by digantdesai — APPROVED** [review]

**Comment by benkli01** [comment]
> > Data layout ops like unsqueeze are not annotated by the quantizer per default which leads to issues down the line
> 
> Can you elaborate? what issues? Is it just lowering issues?

Yes, it's a lowering issue. If the data layout op is in the beginning of the graph the forward propagation of annotations does not work (unless we annotate the inputs explicitly). We tried implementing a backward propagation as well, but sharing specs in the backward direction seems to be not supported, so we went for explicit annotation. Or are we missing something?

File: backends/arm/quantizer/arm_quantizer.py [hunk 2]
```diff
@@ -271,6 +271,7 @@ class ArmQuantizer(Quantizer):
         "mm",
         "cat",
         "one_to_one",
+        "generic",
```
[the remark this answers is not in the document]
Comment (benkli01) [reply backends/arm/quantizer/arm_quantizer.py:274]: No, it is a bit vague (generic even 😅), but naming things is hard...

**Pull request closed without merging**